lightning/src/ln/onion_utils.rs

```diff
@@ -1454,6 +1468,15 @@ impl Hop {
 			_ => false,
 		}
 	}
+
+	pub(crate) fn shared_secret(&self) -> SharedSecret {
+		match self {
+			Hop::Forward { shared_secret, .. } => shared_secret.clone(),
+			Hop::BlindedForward { shared_secret, .. } => shared_secret.clone(),
+			Hop::Receive { shared_secret, .. } => shared_secret.clone(),
+			Hop::BlindedReceive { shared_secret, .. } => shared_secret.clone(),
```

Comment:
nit: doesn't matter much but could return a reference (or dereference) instead of cloning